src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorFactory.java

```diff
@@ -153,12 +159,16 @@ private void changeRandomChunk(Map<K, V> map, PseudoRandom prng, int mapSize) {
       int chunkSize =
           Math.min(prng.closedRange(1, mapSize - chunkOffset), (int) Math.ceil(mapSize / 10.0));
       int elementsChanged = 0;
-      while (elementsChanged < chunkSize) {
-        map.entrySet()
-            .stream()
-            .skip(chunkOffset + elementsChanged)
-            .findFirst()
-            .ifPresent(entry -> mutateElement(entry, map, prng));
+      Iterator<Map.Entry<K, V>> iterator = map.entrySet().iterator();
+      for (int i = 0; i < chunkOffset; i++) {
+        if (iterator.hasNext()) {
+          iterator.next();
+        }
+      }
+
+      while (iterator.hasNext() && elementsChanged < chunkSize) {
+        Map.Entry<K, V> entry = iterator.next();
+        mutateElement(entry, map, prng);
```

Comment:
You can't (safely) change a map while iterating over it in this way: https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#entrySet--:~:text=If%20the%20map%20is%20modified%20while%20an%20iteration%20over%20the%20set%20is%20in%20progress%20(except%20through%20the%20iterator%27s%20own%20remove%20operation%2C%20or%20through%20the%20setValue%20operation%20on%20a%20map%20entry%20returned%20by%20the%20iterator)%20the%20results%20of%20the%20iteration%20are%20undefined.

You can collect a list of keys in a preparatory step and then just call `mutateElement` on the corresponding entry.